### PR TITLE
Feature check issue mutliple statuses

### PR DESCRIPTION
**Pull request by wasabibob**
Updated Check Issue Task to match multiple statuses.  Changed task input property to take a list of strings instead of a string.  Also added an output property to receive the final Jira Issue status that was matched.  So this status can be used later.  This is useful with Jira workflows that have multiple (forked) paths.

### CODE REVIEW

File: src/main/resources/jira/CheckIssue.py [hunk 1]
```diff
@@ -31,8 +31,9 @@
         data = json.loads(response.getResponse())['fields']['status']
         issueStatus = data['name']
         print "\nIssue %s status is %s." % (issueId, issueStatus)
-        if issueStatus == expectedStatus:
-            print "\nThe status of issue %s is 'Resolved'" % (issueId)
+        if (issueStatus in expectedStatusList):
```
Comment (ndebuhr) [inline src/main/resources/jira/CheckIssue.py]: Most styling guides would recommend removing the parentheses here - they are not needed.

File: README.md [hunk 2]
```diff
@@ -43,7 +43,14 @@ Trigger when a new issue appears top of the search results of a JQL query (like
 
 ![CheckIssue](images/CheckIssue.png)
 
-
++ Check Issue (statuses) - Polls Jira to check the status of a Issue. It will complete once the issue is in the correct state
```
Comment (ndebuhr) [inline README.md]: 1. "a Issue" -> "an Issue"
2. We mix use of "state" and "status".  I would recommend using one consistently.

File: src/main/resources/jira/CheckIssue.py [hunk 3]
```diff
@@ -31,8 +31,9 @@
...
+            print "\nThe status of issue %s is %s" % (issueId, issueStatus)
+            finalStatus = issueStatus
             break
         else:
             print "\nIssue is not correct status"
```
Comment (ndebuhr) [inline src/main/resources/jira/CheckIssue.py:39]: Should we change this to some plural case?  "Issue is not set to one of the desired statuses"

File: src/main/resources/synthetic.xml [hunk 4]
```diff
@@ -27,8 +27,9 @@
 
     <type type="jira.CheckIssue" extends="jira.JiraScript">
         <property name="issueId" category="input"/>
-        <property name="expectedStatus" category="input" default=""/>
+        <property name="expectedStatusList" category="input" kind="list_of_string"  description="List one or more expected statuses to check for" default=""/>
```
Comment (ndebuhr) [inline src/main/resources/synthetic.xml:30]: The empty string default strikes me as odd here, but I see how that's carried over from how it was.

**Review by ndebuhr — COMMENTED** [review]
A few recommendations.

File: src/main/resources/jira/CheckIssue.py (hunk 1 above)
Comment (wasabibob) [inline src/main/resources/jira/CheckIssue.py]: updated per review

File: src/main/resources/jira/CheckIssue.py (hunk 3 above)
Comment (wasabibob) [reply src/main/resources/jira/CheckIssue.py:39]: I think it's okay

File: src/main/resources/jira/CheckIssue.py (hunk 1 above)
Comment (wasabibob) [inline src/main/resources/jira/CheckIssue.py]: updated per reveiw

File: src/main/resources/synthetic.xml (hunk 4 above)
Comment (wasabibob) [reply src/main/resources/synthetic.xml:30]: Yeah, I agree. I wasn't sure if it would cause a problem by removing it.

**Review by wasabibob — COMMENTED** [review]
I think this is good now.

**Comment by lionelve** [comment]
Is there a migration path from releases that used single status to this new status list?

**Pull request merged**